### PR TITLE
add onboarding screens

### LINKED ISSUE #144: [Desktop]: add onboarding pages
use figma!

Screenshot from 2023-10-06 11-56-20

### PR DESCRIPTION
**Pull request by Sirmorrison**
Fix #144 

This Pr introduces new onboarding pages.
image

### CODE REVIEW

**Comment by dreacot** [comment]
after going through the onboarding steps i get to this screen below

Screenshot from 2023-10-16 11-27-17

i add a wallet, and on quiting the app and reopening i'm taken to the on boarding screens again, even after completing the onboarding screens again, i'm taken to to the add wallet screen even though i already have a wallet

**Comment by Sirmorrison** [comment]
> after going through the onboarding steps i get to this screen below
> 
> Screenshot from 2023-10-16 11-27-17
> 
> i add a wallet, and on quiting the app and reopening i'm taken to the on boarding screens again, even after completing the onboarding screens again, i'm taken to to the add wallet screen even though i already have a wallet

yes.. that seems to be the flow of the design created and when i asked, @itswisdomagain said this should be the flow. the design has a button for skipping to the home page by clicking the skip wallet creation button.

**Comment by Sirmorrison** [comment]
You originally cannot skip wallet creation if you dont have a wallet already.

**Comment by dreacot** [comment]
> You originally cannot skip wallet creation if you dont have a wallet already.

also skipping wallet creating wasn't part of the original issue, the original issue regarding skipping wallet creation has been closed https://github.com/crypto-power/cryptopower/issues/143

**Comment by dreacot** [comment]
> i add a wallet, and on quiting the app and reopening i'm taken to the on boarding screens again, even after completing the onboarding screens again, i'm taken to to the add wallet screen even though i already have a wallet

from my original comment, the issue i''m outlining is that the onboarding screens always show up with each start of the app. that should not be the case, then the user has created a wallet the onboarding screens should not show up anymore

File: ui/cryptomaterial/icon_gallery.go [hunk 1]
```diff
@@ -197,6 +198,10 @@ func (i *Icons) DefaultIcons() *Icons {
 	i.ChangeNowIcon = NewImage(decredIcons["changenow"])
 	/* End exchange icons */
 
+	i.MiltiWalletIcon = NewImage(decredIcons["miltiWalletIcon"])
```
Comment (dreacot) [inline ui/cryptomaterial/icon_gallery.go:201]: did you mean this to say multiwallet icon?

**Comment by Sirmorrison** [comment]
> > You originally cannot skip wallet creation if you dont have a wallet already.
> 
> also skipping wallet creating wasn't part of the original issue, the original issue regarding skipping wallet creation has been closed #143

I didnt implement skipping wallet creation. Its just a button on the screens.

**Comment by dreacot** [comment]
okay there seems to be some miscommunication so i'd try and clarify

i just built the latest commit on this PR and here are my observations;

- the on boarding pages now work as intended (i.e not showing when there's already a created wallet)

there are still some things to fix;

1. there should be a skip button under the next button, and if the user clicks the skip button, it takes them to the final screen of the onboarding page as seen below

Screenshot from 2023-10-18 10-09-27

2. while discussing with @itswisdomagain we decided to not show the skip wallet creation button as there's not much a user can do without a wallet (asides viewing market rate and proposals) . see this comment https://github.com/crypto-power/cryptopower/issues/143#issuecomment-1762327762

File: ui/page/start_page.go [hunk 2]
```diff
@@ -23,6 +24,13 @@ type (
 	D = layout.Dimensions
 )
 
+type onBoardingScreen struct {
+	onBoardingScreenTitle string
```
Comment (dreacot) [inline ui/page/start_page.go:28]:
```suggestion
	Title string
```

just title is sufficient

`onBoardingScreen.Title`

**Review by dreacot — CHANGES_REQUESTED** [review]

**Comment by dreacot** [comment]
> > okay there seems to be some miscommunication so i'd try and clarify
> > i just built the latest commit on this PR and here are my observations;
> > 
> > * the on boarding pages now work as intended (i.e not showing when there's already a created wallet)
> > 
> > there are still some things to fix;
> > 
> > 1. there should be a skip button under the next button, and if the user clicks the skip button, it takes them to the final screen of the onboarding page as seen below, the original image attached to this PR is correct
> > 
> > Screenshot from 2023-10-18 10-09-27
> > 
> > 2. while discussing with @itswisdomagain we decided to not show the skip wallet creation button as there's not much a user can do without a wallet (asides viewing market rate and proposals) . see this comment [allow users skip wallet creation process #143 (comment)](https://github.com/crypto-power/cryptopower/issues/143#issuecomment-1762327762)
> 
> So to address the first point, on element we had a discussion some time back (i will try to reference it), where @itswisdomagain mentioned that the onboarding screens should always show up and we should prevent skipping it, until the user fully onboards.
> 
> If this is the case, the initial skip button wont be needed because the moment the user onboards, these pages are no longer displayed. Add a skip button offers the first user the ability to skip and not view the informations on the pages. This is why the skip button was removed but it can always be added back.

Yes you are correct, skipping it in that case meant not showing the onboarding screens again.

a use could have gone through the 3 onboarding steps and never actually create a wallet, in that case they are already familiar with it, so they can have the skip button to go straight to where they can add a wallet.

so yes, let's add the skip button back

**Review by dreacot — APPROVED** [review]

**Pull request merged**